src/searchee.ts

```diff
@@ -160,3 +172,106 @@ export async function createSearcheeFromPath(
 		length: totalLength,
 	});
 }
+
+export function getKeyMetaInfo(stem: string, isAnime: boolean): string {
+	const resM = stem.match(RES_STRICT_REGEX)?.groups?.res;
+	const res = resM ? `.${resM}` : "";
+	const sourceM = sourceRegexParse(stem);
+	const source = sourceM ? `.${sourceM}` : "";
+	const groupM = stem.match(RELEASE_GROUP_REGEX)?.groups?.group;
+	if (groupM) {
+		return `${res}${source}-${groupM}`.toLowerCase();
+	}
+	if (!isAnime) {
+		return `${res}${source}`.toLowerCase();
+	}
+	const groupAnimeM = stem.match(ANIME_REGEX)?.groups?.group;
+	if (groupAnimeM) {
+		return `${res}${source}-${groupAnimeM}`.toLowerCase();
+	}
+	return `${res}${source}`.toLowerCase();
+}
+
+export function getMovieKey(stem: string): {
+	ensembleTitle: string;
+	keyTitle: string;
+	year: number;
+} | null {
+	const match = stem.match(MOVIE_REGEX);
+	if (!match) return null;
+	const keyTitle = reformatTitleForSearching(match.groups!.title, false)
+		.replace(NON_UNICODE_ALPHANUM_REGEX, "")
+		.toLowerCase();
+	if (!keyTitle.length) return null;
+	const year = extractInt(match.groups!.year);
+	const ensembleTitle = `${match.groups!.title}.${year}`;
+	return { ensembleTitle, keyTitle, year };
+}
+
+export function getSeasonKey(stem: string): {
+	ensembleTitle: string;
+	keyTitle: string;
+	season: string;
+} | null {
+	const match = stem.match(SEASON_REGEX);
+	if (!match) return null;
+	const keyTitle = reformatTitleForSearching(match.groups!.title, false)
+		.replace(YEAR_REGEX, "")
+		.replace(NON_UNICODE_ALPHANUM_REGEX, "")
+		.toLowerCase();
+	if (!keyTitle.length) return null;
+	const season = `S${extractInt(match.groups!.season)}`;
+	const ensembleTitle = `${match.groups!.title}.${season}`;
+	return { ensembleTitle, keyTitle, season };
+}
+
+export function getEpisodeAndKey(stem: string): {
+	ensembleTitle: string;
+	keyTitle: string;
+	season: string;
+	episode: number | string;
+} | null {
+	const match = stem.match(EP_REGEX);
+	if (!match) return null;
+	const keyTitle = reformatTitleForSearching(match!.groups!.title, false)
+		.replace(YEAR_REGEX, "")
+		.replace(NON_UNICODE_ALPHANUM_REGEX, "")
+		.toLowerCase();
+	if (!keyTitle.length) return null;
+	const seasonM = match!.groups!.season;
+	const season = seasonM
+		? `S${extractInt(seasonM)}`
+		: `S${match!.groups!.year}`;
+	const ensembleTitle = `${match!.groups!.title}.${season}`;
+	const episode = seasonM
+		? extractInt(match!.groups!.episode)
+		: `${match!.groups!.month}/${match!.groups!.day}`;
+	return { ensembleTitle, keyTitle, season, episode };
```

Comment:
why would the ensembleTitle be the same for seasons and episodes?

Reply:
ensembleTitle is only used in the `ensemble-matching` branch to give a good name for the virtual searchee. From rss/announce, we get the title from `getSeasonKey(candidate.name)`. When searching, we get the title from `getEpisodeAndKey(searchee.name)` to name the virtual season searchee.